Kconfig.zephyr

```diff
@@ -250,6 +250,14 @@ config COMPILER_OPT
 	  and can be used to change compiler optimization, warning and error
 	  messages, and so on.
 
+config AVOID_COMPILER_BUILTINS
+	bool "Avoid using compiler builtin functions"
+	help
+	  When this option is set, Zephyr will avoid using the compiler's
+	  builtin functions for extra math operations. Portable C
```

Comment:
> for extra math operations.

The option name implies something else. It implies we would avoid built-in memcmp etc. as well.

Perhaps the option name should be extended to disambiguate?

Reply:
Yeah, I agree.

I'm not even sure if this should be a Kconfig option or just a feature macro (with a long name) that can be defined before including the header. It's really only useful for the unit test which wants to test the portable implementations.